src/components/TimelineComponent.tsx

```diff
@@ -853,6 +855,68 @@ export default class TimelineComponent
         );
   }
 
+  /**
+   * Purpose: updates dataIdx, data, and ordinals when drawing an EventMagnitude
+   * Timeline
+   */
+  getEventMagnitudeData() {
```

Comment:
These JS docs are good the only thing I might consider adding is post conditions so we know how the system changed after we run these